pkg/datamover/datamover.go

```diff
@@ -0,0 +1,120 @@
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"time"
+)
+
+const (
+	ReconciledReasonError = "Error"
+	ConditionReconciled   = "Reconciled"
+)
+
+func GetVolumeSnapMoverClient() (kbclient.Client, error) {
+	client, err := kbclient.New(config.GetConfigOrDie(), kbclient.Options{})
+	if err != nil {
+		return nil, err
+	}
+	err = snapmoverv1alpha1.AddToScheme(client.Scheme())
+	if err != nil {
+		return nil, err
+	}
+
+	return client, err
+}
+
+func DeleteVSBsIfComplete(backupName string) error {
+	volumeSnapMoverClient, err := GetVolumeSnapMoverClient()
+	if err != nil {
+		log.Errorf(err.Error())
+		return err
+	}
+
+	VSBList := snapmoverv1alpha1.VolumeSnapshotBackupList{}
+	VSBListOptions := kbclient.MatchingLabels(map[string]string{
+		velerov1api.BackupNameLabel: backupName,
+	})
+
+	err = volumeSnapMoverClient.List(context.TODO(), &VSBList, VSBListOptions)
+	if err != nil {
+		log.Errorf(err.Error())
+		return err
+	}
+
+	if len(VSBList.Items) > 0 {
+		err = CheckIfVolumeSnapshotBackupsAreComplete(context.Background(), VSBList)
+		if err != nil {
+			log.Errorf("failed to wait for VolumeSnapshotBackups to be completed: %s", err.Error())
+			return err
+		}
+
+		// Now Delete the VSBs
+		for _, vsb := range VSBList.Items {
+			log.Infof("Cleaning up completed VSB: %s/%s", vsb.Namespace, vsb.Name)
+			err := volumeSnapMoverClient.Delete(context.Background(), &vsb)
+			if err != nil {
+				log.Warnf("failed to delete completed VolumeSnapshotBackup: %s", err.Error())
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func CheckIfVolumeSnapshotBackupsAreComplete(ctx context.Context, volumesnapshotbackups snapmoverv1alpha1.VolumeSnapshotBackupList) error {
+	eg, _ := errgroup.WithContext(ctx)
+	// default timeout value is 10
+	timeoutValue := "10m"
+	timeout, err := time.ParseDuration(timeoutValue)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing the datamover timout")
+	}
+	interval := 5 * time.Second
+
+	volumeSnapMoverClient, err := GetVolumeSnapMoverClient()
+	if err != nil {
+		return err
+	}
+
+	for _, vsb := range volumesnapshotbackups.Items {
+		volumesnapshotbackup := vsb
+		eg.Go(func() error {
+			err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+				currentVSB := snapmoverv1alpha1.VolumeSnapshotBackup{}
+				err := volumeSnapMoverClient.Get(ctx, kbclient.ObjectKey{Namespace: volumesnapshotbackup.Namespace, Name: volumesnapshotbackup.Name}, &currentVSB)
+				if err != nil {
+					return false, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshotbackup %s/%s", volumesnapshotbackup.Namespace, volumesnapshotbackup.Name))
+				}
+				// check for a failed VSB
+				for _, cond := range currentVSB.Status.Conditions {
+					if cond.Status == metav1.ConditionFalse && cond.Reason == ReconciledReasonError && cond.Type == ConditionReconciled {
+						return false, errors.Errorf("volumesnapshotbackup %s has failed status", currentVSB.Name)
+					}
+				}
```

Comment:
Could also add in a check for VSB phase being failed or partiallyFailed. Because there are instances where conditions will be false, such as the objectModified error, but the controller will reconcile and have true conditions again.